src/views/CourtOrder.vue

```diff
@@ -0,0 +1,696 @@
+      // route to dashboard
+      this.navigateToBusinessDashboard(this.getIdentifier)
+      return
+    }
+
+    // open confirmation dialog and wait for response
+    this.$refs.confirm.open(
+      'Unsaved Changes',
+      'You have unsaved changes in your Continue Out. Do you want to exit your filing?',
+      {
+        width: '45rem',
+        persistent: true,
+        yes: 'Return to my filing',
+        no: null,
+        cancel: 'Exit without saving'
+      }
+    ).then(() => {
+      // if we get here, Yes was clicked
+      // nothing to do
+    }).catch(() => {
+      // if we get here, Cancel was clicked
+      // ignore changes
+      this.haveChanges = false
+      // route to dashboard
+      this.navigateToBusinessDashboard(this.getIdentifier)
+    })
+  }
+
+  @Watch('notation')
+  async onNotationChanged (): Promise<void> {
+    // if this is a court order and notation has changed, re-validate file upload component
+    if (this.isCourtOrder && this.enableValidation) {
+      await this.$nextTick() // wait for variables to update
+      this.$refs.fileUploadRef?.validate()
+    }
+  }
+
+  // NB: watch "fileKey" because it changes according to file validity (while "file" might not)
+  @Watch('fileKey')
+  async onFileKeyChanged (): Promise<void> {
+    // if this is a court order and file has changed, re-validate notation form
+    if (this.isCourtOrder && this.enableValidation) {
+      await this.$nextTick() // wait for variables to update
+      this.$refs.notationFormRef?.validate()
+    }
+  }
+
+  openFileDialog () {
+    // Access the underlying file input element and trigger click
+    const fileInput = this.$refs.fileUploadRef.$el.querySelector('input')
+    if (fileInput) {
+      fileInput.click() // Open file input dialog
+    }
+  }
+
+  onFileChange (file) {
+    this.file = file // Update the file data in the parent component
+  }
+
+  readonly validComponents = [
+    'court-order-section'
+  ]
+
+  get validFlags (): object {
+    return {
+      courtOrder: this.courtOrderValid
+    }
+  }
+
+  @Watch('staffPaymentValid')
+  @Watch('courtOrderValid')
+  onHaveChanges (): void {
+    this.haveChanges = true
+  }
+
+    @Watch('staffPaymentData')
+  onStaffPaymentDataChanged (val: StaffPaymentIF): void {
+    const waiveFees = (val.option === StaffPaymentOptions.NO_FEE)
+
+    // add Waive Fees flag to all filing codes
+    this.updateFilingData('add', FilingCodes.COURT_ORDER, val.isPriority, waiveFees)
+
+    this.haveChanges = true
+  }
+  @Watch('dialog')
+    onDialogChanged (val: boolean): void {
+    // when dialog is hidden, reset everything
+      if (!val) {
+      // disable validation
+        this.enableValidation = false
+
+        // reset notation form
+        this.$refs.notationFormRef?.reset()
+
+        // reset file upload component
+        this.$refs.fileUploadRef?.reset()
+
+        // reset court order component and variables
+        this.courtOrderPoaKey++ // force re-render
```

Comment:
Thanks for the comment. We don't do this in other places but I think this is OK.